script.js

```diff
@@ -1,4 +1,212 @@
+    console.log(`Check current game mode`);
+    console.log(currentMode);
+  } else if (currentMode == `P1` && input == "" && p1Counter == 0) {
+    // Player 1 begins play
+    myOutputValue = playBeatThatP1(input);
+    pergameCounter = pergameCounter + 1;
+    console.log(`counter is`);
+    console.log(pergameCounter);
+    p1Counter = p1Counter + 1;
+  } else if (currentMode == `P2` && input == "" && p2Counter == 0) {
+    // Player 2 begins play
+    myOutputValue = playBeatThatP2(input);
+    pergameCounter = pergameCounter + 1;
+    console.log(`counter is`);
+    console.log(pergameCounter);
+    p2Counter = p2Counter + 1;
+  } else if (
+    // Player chooses dice position and the final number is stored in global variable
+    (currentMode == `P1` && (input == "D1" || input == "D2")) ||
+    (currentMode == `P2` && (input == "D1" || input == "D2"))
+  ) {
+    var playerfinalNum = createNumber(input);
+    // final number has been generated but now need to put final number in array
+    if (currentMode == "P1") {
+      p1score = playerfinalNum;
+    } else if (currentMode == "P2") {
+      p2score = playerfinalNum;
+    }
+    myOutputValue = `Hi ${currentMode}, your final score is ${playerfinalNum}. 🎉 <br> <br> Type "Next" in the box and hit submit to continue the game. ▶`;
+  } else if (input == "Next" && pergameCounter == 1) {
+    myOutputValue = remainingplayerChecker(input);
+  } else if (pergameCounter == 2) {
+    winnerResult = winnerChecker(input);
+    console.log(winnerResult);
+    myOutputValue = `${gameWinner}.🤩 <br> <br> Player 1 scored ${p1score} and player 2 scored ${p2score}.`;
+  } else if (
+    (p1Counter == 1 && input == "") ||
+    (p2Counter == 1 && input == "")
+  ) {
+    //Players cannot play more than once within a game
+    myOutputValue = `Sorry ${currentMode} you already played.💢💢 You can only roll once per game. <br><br> Please type "Next" and hit submit to continue the game. ▶`;
+  }
+
   return myOutputValue;
 };
+
+// ============================================ MAIN FUNCTION ENDS HERE =================================================================
+
+// ------------------------------------------HELPER FUNCTIONS BELOW --------------------------
+
+// Beat that function for player 1
+var playBeatThatP1 = function () {
+  var diceCounter = 0;
+  console.log(`current dice counter is`);
+  console.log(diceCounter);
+  while (diceCounter < numberofDice) {
+    // call dice roll function into the loop
+    var diceResult = rollDice();
+    // add dice rolls to player1
+    player1scores.push(diceResult);
+    diceCounter = diceCounter + 1;
+    console.log(`dice count is now`);
+    console.log(diceCounter);
+  }
+
+  console.log(`check player 1 score`);
+  console.log(player1scores);
+  return `You rolled: ${player1scores}. <br><br>Now please enter "D1" or "D2" to choose which number goes first in your final score.`;
+};
+
+// Beat that function for player 2
+var playBeatThatP2 = function () {
```

Comment:
You can consider combining `playBeatThatP1` and `playBeatThatP2` into a single function